redis/asyncio/cluster.py

```diff
@@ -1331,13 +1333,17 @@ def set_nodes(
         if remove_old:
             for name in list(old.keys()):
                 if name not in new:
-                    task = asyncio.create_task(old.pop(name).disconnect())  # noqa
+                    task = asyncio.create_task(old.pop(name).disconnect())
+                    self._background_tasks.add(task)
+                    task.add_done_callback(self._background_tasks.discard)
```

Comment:
Fixed an issue where if multiple tasks were created, all but the last one would no longer be strongly referenced. Tasks that are not strongly referenced can be deleted by the GC.